cli/Cargo.toml

```diff
@@ -11,6 +11,13 @@ license.workspace = true
 repository.workspace = true
 description = "Provides the deno executable"
 
+[features]
+default = ["upgrade"]
+# A feature that enables the upgrade subcommand and the background check for
+# available updates (of deno binary). This is typically disabled for (Linux)
+# distribution packages.
+upgrade = []
```

Comment:
Instead of maintaining conditional compilation, perhaps`deno upgrade` should instead recommend running the appropriate command to upgrade? For example, in dprint I check the path of the executable and provide upgrade instructions https://github.com/dprint/dprint/blob/a2d9751c02624f1ab88dd6feb39e4984ff75bd4f/crates/dprint/src/commands/upgrade.rs#L36

Reply:
I think path sniffing will always be a cat-and-mouse game,
given that Deno cannot know how exactly an install path is used.

Using a compile flag will not work for package managers that just download the release build from GitHub unless an extra artifact is provided. Although I am not aware of one that does this.

Python [solved](https://peps.python.org/pep-0668/) a loosely similar problem by using a marker file in a specific location, but they have a the convenience of a stdlib path for this.